zen/ui/nodes/power-button.c

```diff
@@ -0,0 +1,169 @@
+#include "zen/ui/nodes/power-button.h"
+
+#include <cairo.h>
+#include <zen-common.h>
+#include <zigzag.h>
+
+#include "zen/server.h"
+
+double icon_width = 10.;
+double icon_height = 10.;
+
+static int
+zn_power_button_handle_second_timer(void *data)
+{
+  struct zn_power_button *self = data;
+  struct zigzag_node *zigzag_node = self->zigzag_node;
+  struct zn_server *server = zn_server_get_singleton();
+
+  int64_t time_ms = current_realtime_clock_ms();
+  self->next_sec_ms = (time_ms - time_ms % MSEC_PER_SEC + MSEC_PER_SEC);
+
+  int ms_delay = (int)(self->next_sec_ms - time_ms + 10);
+  wl_event_source_timer_update(self->second_timer_source, ms_delay);
+
+  zigzag_node_update_texture(zigzag_node, server->renderer);
+  return 0;
+}
+
+static void
+zn_power_button_on_click(struct zigzag_node *self, double x, double y)
+{
+  UNUSED(self);
+  UNUSED(x);
+  UNUSED(y);
+  // TODO: Show "Logout"
+}
+
+static bool
+zn_power_icon_render(struct zigzag_node *self, cairo_t *cr)
+{
+  UNUSED(self);
+  cairo_set_source_rgba(cr, 1.0, 1.0, 1.0, 0.7);
+  cairo_paint(cr);
+  bool result = zigzag_cairo_stamp_svg_on_surface(
+      cr, POWER_BUTTON_ICON, 0., 0., icon_width, icon_height);
+  if (!result) {
+    return false;
+  }
+  return true;
+}
+
+static bool
+zn_power_button_render(struct zigzag_node *self, cairo_t *cr)
```

Comment:
In this file, `self` is `zn_power_button`. Sorry for missing in previous review, but please fix it in several methods like so.
```suggestion
zn_power_button_render(struct zigzag_node *node, cairo_t *cr)
```

Reply:
Thanks, I resolved all `zigzag_node *self` arguments in `zen/ui/nodes` to `node`.